phoenix-core/src/main/java/org/apache/phoenix/jdbc/PhoenixStatement.java

```diff
@@ -417,11 +504,22 @@ private PhoenixResultSet executeQuery(final CompilableStatement stmt,
                                                     e.getSchemaName(), e.getTableName(), true)
                                             .wasUpdated()) {
                                         //TODO we can log retry count and error for debugging in LOG table
-                                        return executeQuery(stmt, false, queryLogger, noCommit);
+                                        return executeQuery(stmt, false, queryLogger, noCommit, validateDDLTimestamp);
                                     }
                                 }
                                 throw e;
-                            } catch (RuntimeException e) {
+                            }
+                            catch (SQLException e) {
+                                // force update cache in case of StaleMetadataCacheException and retry
```

Comment:
Trying to understand why we have retries here and also in validateDDLTimestamp? Can we consolidate them just in 1 place? Preferably in validateDDLTimestamp?

Reply:
As discussed, validateDDLTimestamp retry is just for the RPC call. This retry is for executeQuery() after force-updating the client cache.